rb/lib/selenium/webdriver/common/fedcm/dialog.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Selenium
+  module WebDriver
+    module FedCM
+      class Dialog
+        def initialize(bridge)
+          @bridge = bridge
+        end
+
+        DIALOG_TYPE_ACCOUNT_LIST = 'AccountChooser'
+        DIALOG_TYPE_AUTO_REAUTH = 'AutoReauthn'
+
+        # Closes the dialog as if the user had clicked X.
+        def click
+          @bridge.click_fedcm_dialog_button
+        end
+
+        # Closes the dialog as if the user had clicked X.
+        def cancel
+          @bridge.cancel_fedcm_dialog
+        end
+
+        # Selects an account as if the user had clicked on it.
+        #
+        # @param [Integer] index The index of the account to select from the list returned by get_accounts.
```

Comment:
Should this accept `Account` as a parameter instead? This would make the API more user-friendly:


```ruby
accounts = driver.fedcm_dialog.accounts
driver.fedcm_dialog.select_accounts(accounts.first)
```

Reply:
The thing is that when checking how is described on the docs https://fedidcg.github.io/FedCM/#webdriver-selectaccount:

The [remote end steps](https://w3c.github.io/webdriver/#dfn-remote-end-steps) are:

If parameters is not a JSON [Object](https://tc39.es/ecma262/multipage/structured-data.html#sec-json-object), return a [WebDriver error](https://w3c.github.io/webdriver/#dfn-error) with [error code](https://w3c.github.io/webdriver/#dfn-error-code) [invalid argument](https://w3c.github.io/webdriver/#dfn-invalid-argument).

If no FedCM dialog is currently open, return a [WebDriver error](https://w3c.github.io/webdriver/#dfn-error) with [error code](https://w3c.github.io/webdriver/#dfn-error-code) [no such alert](https://w3c.github.io/webdriver/#dfn-no-such-alert).

Let accountIndex be the result of [getting a property](https://w3c.github.io/webdriver/#dfn-getting-properties) named "accountIndex" from parameters.

If accountIndex is [undefined](https://webidl.spec.whatwg.org/#idl-undefined) or is less than 0 or greater than or equal to the number of accounts that the user can choose from in the current flow, return a [WebDriver error](https://w3c.github.io/webdriver/#dfn-error) with [error code](https://w3c.github.io/webdriver/#dfn-error-code) [invalid argument](https://w3c.github.io/webdriver/#dfn-invalid-argument).

Close the dialog and continue the [create an IdentityCredential](https://fedidcg.github.io/FedCM/#create-an-identitycredential) algorithm as if the user had selected the account indicated by accountIndex and [granted permission to sign-up](https://fedidcg.github.io/FedCM/#request-permission-to-sign-up), if applicable.

Return [success](https://w3c.github.io/webdriver/#dfn-success) with data null.

So if we would like to provide the user the opportunity to pass an account our method should look like this:

`def select_account(account)
       account_index = accounts.index(account)
      @bridge.select_fedcm_account account_index
  end
  `
  
  Is that something we want to do? because then we will also need error handling to clarify that the account is not present

Reply:
Ok, let's keep using `index` instead - that's also what Java does, we can provide convenient API later.